torch/_dynamo/repro/after_dynamo.py

```diff
@@ -184,7 +185,72 @@ def generate_dynamo_fx_repro_string(
     )
 
 
-def dump_backend_repro_as_file(gm, args, compiler_name, check_accuracy=False):
+def generate_dynamo_fx_test_from_repro_string(
+    gm,
+    args,
+    compiler_name,
+    check_accuracy=False,
+    *,
+    stable_output=False,
+    save_dir=None,
+    command="run",
+):
+    """
+    Generate a test string for backend-agnostic minified version.
+    """
+
+    model_str = NNModuleToString.convert(gm, imports=False)
+
+    # TODO: Figure out why torch.compile'd hash isn't work on this codepath
+    writer = TestInputWriter(save_dir, stable_hash=True)
+    for placeholder, arg in zip(fx_placeholder_targets(gm), args):
+        if isinstance(arg, (int, torch.SymInt)):
+            writer.symint(placeholder, arg)
+        elif isinstance(arg, torch.Tensor):
+            # TODO: improve these names with FQN
+            writer.tensor(placeholder, arg)
+        else:
+            raise TypeError(f"arg is neither SymInt/int nor torch.Tensor, {arg}")
+    load_args = "\n".join(writer.lines())
+    storage_names = ",".join(writer._storage_names)
+
+    return textwrap.dedent(
+        f"""
+from math import inf
+import torch
+from torch import tensor, device
+import torch.fx as fxindent
+import torch._dynamo
+from torch._dynamo.testing import rand_strided
+from torch._dynamo.debug_utils import run_fwd_maybe_bwd
+from torch.nn import *
+import torch._dynamo.test_case
+import torch._dynamo.testing
+
+{generate_config_string(stable_output=stable_output)}
+
+{extra_imports}
+
+class GeneratedReproTestCase(torch._dynamo.test_case.TestCase):
+    def test_generated_repro_case(self):
```

Comment:
Don't you want the configs patched at the test case, rather than at the top of the file. Since ostensibly you're going to put the test case into one of the preexisting test case files?

Reply:
>  Since ostensibly you're going to put the test case into one of the preexisting test case files?

No, I assumed you would just generate a test here for the person to copy/paste. I do not want to pollute or change existing code as part of this, that could make for ugly commits or unintended writes.

Reply:
Oh you mean like, generate @patch config statements? Yeah, we can do that.